src/uct/ib/ud/accel/ud_mlx5.c

```diff
@@ -821,7 +821,6 @@ static UCS_CLASS_CLEANUP_FUNC(uct_ud_mlx5_iface_t)
     uct_ud_iface_remove_async_handlers(&self->super);
     uct_ud_enter(&self->super);
     UCT_UD_IFACE_DELETE_EPS(&self->super, uct_ud_mlx5_ep_t);
-    ucs_twheel_cleanup(&self->super.async.slow_timer);
```

Comment:
seems it has to be free somewhere, I see leaks with ASan when running it locally:
```
Direct leak of 32768 byte(s) in 2 object(s) allocated from:
    #0 0x7ffff72b9f48 in __interceptor_malloc ../../../../source/gcc-9.2.0/libsanitizer/asan/asan_malloc_linux.cc:144
    #1 0x7ffff6e423b1 in ucs_malloc debug/memtrack.c:182
    #2 0x7ffff6e66644 in ucs_twheel_init time/timer_wheel.c:25
    #3 0x7ffff5d3cca3 in uct_ud_iface_complete_init ud/base/ud_iface.c:356
    #4 0x7ffff5d6a782 in uct_ud_verbs_iface_t_init ud/verbs/ud_verbs.c:670
    #5 0x7ffff5d6b28b in uct_ud_verbs_iface_t_new ud/verbs/ud_verbs.c:684
    #6 0x7ffff6afaf8d in uct_iface_open base/uct_md.c:255
    #7 0x7d8de1 in uct_test::entity::entity(resource const&, uct_iface_config*, uct_iface_params*, uct_md_config*) uct/uct_test.cc:765
    #8 0x7d6c29 in uct_test::create_entity(unsigned long, ucs_status_t (*)(void*, uct_ep*, ucs_status_t), ucs_status_t (*)(void*, void*, unsigned long, unsigned int, unsigned long, unsigned long, void**), ucs_status_t (*)(void*, unsigned
 int, unsigned long, void const*, unsigned int, unsigned long, unsigned long, void const*), void*, void*, ucs_status_t (*)(void*, unsigned int), void*) uct/uct_test.cc:647
    #9 0xd93b74 in test_uct_ib::create_connected_entities() uct/ib/test_ib.cc:13
    #10 0xd93d51 in test_uct_ib::init() uct/ib/test_ib.cc:24
    #11 0xda7cd3 in test_uct_event_ib::init() uct/ib/test_ib.cc:438
    #12 0x5d37bf in ucs::test_base::SetUpProxy() common/test.cc:242
    #13 0x61b2f5 in uct_test::SetUp() uct/uct_test.h:96
    #14 0x57e98a in void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (/auto/mtrswgwork/dmitrygla/ucx/test/gtest/gtest+0x57e98a)
    #15 0x5737a3 in void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (/auto/mtrswgwork/dmitrygla/ucx/test/gtest/gtest+0x5737a3)
    #16 0x5340a9 in testing::Test::Run() common/gtest-all.cc:3630
    #17 0x535225 in testing::TestInfo::Run() common/gtest-all.cc:3812
    #18 0x535d51 in testing::TestCase::Run() common/gtest-all.cc:3930
    #19 0x545e18 in testing::internal::UnitTestImpl::RunAllTests() common/gtest-all.cc:5808
    #20 0x581987 in bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (/auto/mtrswgwork/dmitr
ygla/ucx/test/gtest/gtest+0x581987)
    #21 0x575b3b in bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (/auto/mtrswgwork/dmitrygl
a/ucx/test/gtest/gtest+0x575b3b)
    #22 0x543322 in testing::UnitTest::Run() common/gtest-all.cc:5419
    #23 0x59b059 in RUN_ALL_TESTS() common/gtest.h:20059
    #24 0x59adad in main common/main.cc:102
    #25 0x7ffff4128c04 in __libc_start_main (/usr/lib64/libc.so.6+0x21c04)
```
```
Direct leak of 32768 byte(s) in 2 object(s) allocated from:
    #0 0x7ffff72b9f48 in __interceptor_malloc ../../../../source/gcc-9.2.0/libsanitizer/asan/asan_malloc_linux.cc:144
    #1 0x7ffff6e423b1 in ucs_malloc debug/memtrack.c:182
    #2 0x7ffff6e66644 in ucs_twheel_init time/timer_wheel.c:25
    #3 0x7ffff5d3cca3 in uct_ud_iface_complete_init ud/base/ud_iface.c:356
    #4 0x7ffff5d8de45 in uct_ud_mlx5_iface_t_init ud/accel/ud_mlx5.c:820
    #5 0x7ffff5d8e97e in uct_ud_mlx5_iface_t_new ud/accel/ud_mlx5.c:841
    #6 0x7ffff6afaf8d in uct_iface_open base/uct_md.c:255
    #7 0x7d8de1 in uct_test::entity::entity(resource const&, uct_iface_config*, uct_iface_params*, uct_md_config*) uct/uct_test.cc:765
    #8 0x7d6c29 in uct_test::create_entity(unsigned long, ucs_status_t (*)(void*, uct_ep*, ucs_status_t), ucs_status_t (*)(void*, void*, unsigned long, unsigned int, unsigned long, unsigned long, void**), ucs_status_t (*)(void*, unsigned
 int, unsigned long, void const*, unsigned int, unsigned long, unsigned long, void const*), void*, void*, ucs_status_t (*)(void*, unsigned int), void*) uct/uct_test.cc:647
    #9 0xd93b0c in test_uct_ib::create_connected_entities() uct/ib/test_ib.cc:12
    #10 0xd93d51 in test_uct_ib::init() uct/ib/test_ib.cc:24
    #11 0xda7cd3 in test_uct_event_ib::init() uct/ib/test_ib.cc:438
    #12 0x5d37bf in ucs::test_base::SetUpProxy() common/test.cc:242
    #13 0x61b2f5 in uct_test::SetUp() uct/uct_test.h:96
    #14 0x57e98a in void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (/auto/mtrswgwork/dmitrygla/ucx/test/gtest/gtest+0x57e98a)
    #15 0x5737a3 in void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (/auto/mtrswgwork/dmitrygla/ucx/test/gtest/gtest+0x5737a3)
    #16 0x5340a9 in testing::Test::Run() common/gtest-all.cc:3630
    #17 0x535225 in testing::TestInfo::Run() common/gtest-all.cc:3812
    #18 0x535d51 in testing::TestCase::Run() common/gtest-all.cc:3930
    #19 0x545e18 in testing::internal::UnitTestImpl::RunAllTests() common/gtest-all.cc:5808
    #20 0x581987 in bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (/auto/mtrswgwork/dmitrygla/ucx/test/gtest/gtest+0x581987)
    #21 0x575b3b in bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (/auto/mtrswgwork/dmitrygla/ucx/test/gtest/gtest+0x575b3b)
    #22 0x543322 in testing::UnitTest::Run() common/gtest-all.cc:5419
    #23 0x59b059 in RUN_ALL_TESTS() common/gtest.h:20059
    #24 0x59adad in main common/main.cc:102
    #25 0x7ffff4128c04 in __libc_start_main (/usr/lib64/libc.so.6+0x21c04)
```

Reply:
Oops, missed this.
I thought to clean it in UCS_CLASS_CLEANUP_FUNC(uct_ud_iface_t),

Reply:
> I thought to clean it in UCS_CLASS_CLEANUP_FUNC(uct_ud_iface_t),

@Artemy-Mellanox makes sense 👍 thank you!